crates/core/src/routing_info/mod.rs

```diff
@@ -0,0 +1,219 @@
+/// herd of queries to the source of truth.
+//
+// Implementation note: the sole authority for node-level routing is currently the metadata store,
+// and in particular, the cluster scheduling plan. This will change in the future so avoid leaking
+// implementation details or assumptions about the source of truth.
+pub struct PartitionRoutingRefresher {
+    sender: CommandSender,
+    receiver: CommandReceiver,
+    metadata_store_client: MetadataStoreClient,
+    inflight_refresh_task: Option<TaskHandle<anyhow::Result<()>>>,
+    inner: Arc<ArcSwap<PartitionToNodesRoutingTable>>,
+}
+
+impl PartitionRoutingRefresher {
+    pub fn new(metadata_store_client: MetadataStoreClient) -> Self {
+        let (sender, receiver) = mpsc::channel(1);
+        Self {
+            metadata_store_client,
+            receiver,
+            sender,
+            inflight_refresh_task: None,
+            inner: Arc::new(ArcSwap::new(Arc::new(PartitionToNodesRoutingTable {
+                inner: HashMap::default(),
+            }))),
+        }
+    }
+
+    /// Get a handle to the partition-to-node routing table.
+    pub fn partition_routing(&self) -> PartitionRouting {
+        PartitionRouting {
+            sender: self.sender.clone(),
+            partition_to_node_mappings: self.inner.clone(),
+        }
+    }
+
+    async fn run(mut self) -> anyhow::Result<()> {
+        debug!("Metadata manager started");
+
+        let update_interval = Configuration::pinned()
+            .common
+            .metadata_update_interval
+            .into();
+        let mut update_interval = tokio::time::interval(update_interval);
+        update_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        let mut cancel = pin!(cancellation_watcher());
+
+        loop {
+            tokio::select! {
+                _ = &mut cancel => {
+                    info!("Routing information refresher stopped");
+                    break;
+                }
+                Some(cmd) = self.receiver.recv() => {
+                    match cmd {
+                        Command::SyncRoutingInformation(target_version) => {
+                            self.spawn_sync_routing_information_task(target_version);
+                        }
+                    }
+                }
+                _ = update_interval.tick() => {
+                    debug!("Refreshing routing information...");
+                    self.spawn_sync_routing_information_task(TargetVersion::Latest);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn spawn_sync_routing_information_task(&mut self, target_version: TargetVersion) {
+        if !self
+            .inflight_refresh_task
+            .as_ref()
+            .is_some_and(|t| !t.is_finished())
+        {
+            let partition_to_node_mappings = self.inner.clone();
+            let metadata_store_client = self.metadata_store_client.clone();
+
+            let task = task_center().spawn_unmanaged(
```

Comment:
I use `spawn_unmanaged` to get a task handle which allows me to check if the previous one is finished.

Reply:
Good idea 👍